PhysicsTools/PatAlgos/python/producersHeavyIons/heavyIonJetSetup.py

```diff
@@ -0,0 +1,229 @@
+        ak5HiGenJets.clone(rParam = radius / 10),
+        process,
+        task)
+
+    addToProcessAndTask(
+        'ak' + str(radius) + 'HiCleanedGenJets',
+        heavyIonCleanedGenJets.clone(src = genjetcollection),
+        process,
+        task)
+
+    modules = {
+        'patJetGenJetMatch':
+        patJetGenJetMatch.clone(
+            matched = 'ak' + str(radius) + 'HiCleanedGenJets',
+            maxDeltaR = radius / 10,
+            resolveByMatchQuality = True,
+            src = tag + "Jets",
+            ),
+
+        'patJetPartonMatch':
+        patJetPartonMatch.clone(
+            matched = "cleanedPartons",
+            src = tag + "Jets",
+            ),
+
+        'patJetPartons':
+        patJetPartons.clone(
+            particles = "hiSignalGenParticles",
+            ),
+
+        'patJetFlavourAssociation':
+        patJetFlavourAssociation.clone(
+            jets = tag + "Jets",
+            rParam = radius / 10,
+            bHadrons = "patJetPartons" + tag + ":bHadrons",
+            cHadrons = "patJetPartons" + tag + ":cHadrons",
+            leptons = "patJetPartons" + tag + ":leptons",
+            partons = "patJetPartons" + tag + ":physicsPartons",
+            ),
+
+        'patJetPartonAssociationLegacy':
+        patJetPartonAssociationLegacy.clone(
+            jets = tag + "Jets",
+            partons = "allPartons",
+            ),
+
+        'patJetFlavourAssociationLegacy':
+        patJetFlavourAssociationLegacy.clone(
+            srcByReference = "patJetPartonAssociationLegacy" + tag,
+            ),
+        }
+
+    for label, module in six.iteritems(modules):
+        addToProcessAndTask(label + tag, module, process, task)
+
+def useCsJetsForPat(process, tag):
+    process.patJets.useLegacyJetMCFlavour = True
+    process.patJets.userData.userFloats.src = []
+    process.patJets.userData.userInts.src = []
+    process.patJets.jetSource = tag + "Jets"
+    process.patJets.genJetMatch = "patJetGenJetMatch" + tag
+    process.patJets.genPartonMatch = "patJetPartonMatch" + tag
+    process.patJets.JetFlavourInfoSource = "patJetFlavourAssociation" + tag
+    process.patJets.JetPartonMapSource = "patJetFlavourAssociationLegacy" + tag
+    process.patJets.jetCorrFactorsSource = ["patJetCorrFactors" + tag]
+    process.patJets.trackAssociationSource = "JetTracksAssociatorAtVertex" + tag
+    process.patJets.useLegacyJetMCFlavour = True
+    process.patJets.discriminatorSources = [
+        "SimpleSecondaryVertexHighEffBJetTags" + tag,    
+        "SimpleSecondaryVertexHighPurBJetTags" + tag,
+        "CombinedSecondaryVertexBJetTags" + tag,
+        "CombinedSecondaryVertexV2BJetTags" + tag,
+        "JetBProbabilityBJetTags" + tag,
+        "JetProbabilityBJetTags" + tag,
+        "TrackCountingHighEffBJetTags" + tag,
+        "TrackCountingHighPurBJetTags" + tag,
+    ]
+    process.patJets.tagInfoSources = cms.VInputTag(["ImpactParameterTagInfos" + tag,"SecondaryVertexTagInfos"+tag])
```

Comment:
are the tagInfos really needed?
It's been a while since I've seen tagInfos persisted.

Reply:
Yes, we are looping over these TagInfos in our jet analysis code:  
https://github.com/CmsHI/cmssw/blob/forest_CMSSW_10_3_1/HeavyIonsAnalysis/JetAnalysis/src/HiInclusiveJetAnalyzer.cc